modes.go

```diff
@@ -412,12 +407,29 @@ func DoReadsFromTable(table string, session *gocql.Session, threadResult *result
 	default:
 		request = fmt.Sprintf("SELECT %s FROM %s.%s WHERE pk = ? AND ck >= ? %s LIMIT %d", selectFields, keyspaceName, table, orderBy, rowsPerRequest)
 	}
+	return session.Query(request)
+}
 
-	query := session.Query(request)
+func DoReadsFromTable(table string, session *gocql.Session, threadResult *results.TestThreadResult, workload WorkloadGenerator, rateLimiter RateLimiter) {
+	var queries []*gocql.Query
+	queryIdx := 0
+	queryIdxMax := len(selectOrderByParsed) - 1
+
+	for _, orderBy := range selectOrderByParsed {
+		queries = append(queries, BuildReadQuery(table, orderBy, session))
+	}
 
 	RunTest(threadResult, workload, rateLimiter, func(rb *results.TestThreadResult) (error, time.Duration) {
 		pk := workload.NextPartitionKey()
 
+		query := queries[queryIdx]
+
+		if queryIdx >= queryIdxMax {
+			queryIdx = 0
+		} else {
+			queryIdx++
+		}
```

Comment:
I don't think this is correct. queryIdx needs to be first incremented and then checked if it's not >= queryIdxMax.
At the moment `queryIdx++` can increment queryIdx to queryIdxMax and the next iteration will try to access queries table with index equal to queryIdxMax.

Reply:
queryIdxMax is last valid index, so if you increase it first, you wont' visit last element.
 Also increasing it first gives you 3 steps on worst case scenario, while current code gives you 2.

Reply:
right. it works then. my bad